src/api/routes/v1/amendments.ts

```diff
@@ -11,45 +12,301 @@ interface AmendmentsInfoResponse {
   amendments: AmendmentsInfo[]
 }
 
+interface AmendmentsVoteResponse {
+  result: 'success' | 'error'
+  count: number
+  amendments: Array<AmendmentsEnabled | AmendmentInVoting>
+}
 interface AmendmentInfoResponse {
   result: 'success' | 'error'
   amendment: AmendmentsInfo
 }
 
-interface Cache {
+interface CacheInfo {
   amendments: AmendmentsInfo[]
   time: number
 }
 
+interface AmendmentInVoting extends AmendmentsInfo {
+  threshold: string
+  consensus: string
+  voted: {
+    count: number
+    validators: Array<{
+      signing_key: string
+      ledger_index: string
+      unl: boolean
+    }>
+  }
+}
+
+interface AmendmentInVotingMap {
+  [key: string]: {
+    name: string
+    rippled_version: string | undefined | null
+    threshold: string
+    consensus: string
+    validators: Array<{
+      signing_key: string
+      ledger_index: string
+      unl: boolean
+    }>
+    deprecated: boolean | null
+  }
+}
+
+interface AmendmentsEnabled extends AmendmentsInfo {
+  ledger_index?: string
+  tx_hash?: string
+  date?: Date
+}
+
+interface BallotAmendmentDb {
+  signing_key: string
+  ledger_index: string
+  amendments: string
+  unl?: boolean
+}
+
+interface CacheVote {
+  networks: Map<string, Array<AmendmentsEnabled | AmendmentInVoting>>
+  time: number
+}
+
 const log = logger({ name: 'api-amendments' })
 
-const cache: Cache = {
+const cacheInfo: CacheInfo = {
   amendments: [],
   time: Date.now(),
 }
 
+const cacheVote: CacheVote = {
+  networks: new Map<string, Array<AmendmentsEnabled | AmendmentInVoting>>(),
```

Comment:
Should this be ` Map<number, Array<AmendmentsEnabled | AmendmentInVoting>>()`?

Reply:
id for networks on VHS are string (either "main", "dev", or "1", "2", etc.)